roxie/roxiepipe/roxiepipe.cpp

```diff
@@ -677,7 +677,7 @@ int main(int argc, char *argv[])
             {
 #ifdef _USE_OPENSSL
                 if (useSSL)
-                    smartSocketFactory = createSecureSmartSocketFactory(hosts.str(), retryMode);
+                    smartSocketFactory = createSecureSmartSocketFactory(hosts.str(), createSecureSocketConfig(nullptr, nullptr, nullptr), retryMode);
```

Comment:
This is what I expect but it may not be clear, and I wasn't sure if other changes would make it better. I thought this could be a placeholder until the usage of the smart socket classes is regularized. If it would be better to have an empty rather than null object here I could adjust the `createSecureSocketConfig` to do so or create an alternate function. I can see how this approach may not be clear and will adjust if you'd like.

The use of the smart sockets and factories isn't entirely consistent. In some cases the users of the classes interrogate the base interface for TLS status, and in others the user knows the TLS state and uses the correct secure/insecure flavor based on its own knowledge. roxiepipe is in the second case and it doesn't rely on the smart socket class itself having a valid config, rather it has already created a secure socket version based on the ssl argument (and compile switch).